.github/workflows/maven-publish-snapshot.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish SNAPSHOT artifacts to Maven snapshot repository.
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 5 * * Mon-Fri'
+
+jobs:
+  build-and-publish-snapshots:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Checkout Data Prepper
+        uses: actions/checkout@v4
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+
+      - name: Publish snapshots to Maven
+        run: |
+          export SONATYPE_USERNAME=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text)
+          export SONATYPE_PASSWORD=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
+          echo "::add-mask::$SONATYPE_USERNAME"
+          echo "::add-mask::$SONATYPE_PASSWORD"
```

Comment:
We won't need to assume the role and export the env variables for sonatype credentials anymore. We now use `onepassword` to load new credentials. 
```
      - name: Load secret
        uses: 1password/load-secrets-action@v2
        with:
          # Export loaded secrets as environment variables
          export-env: true
        env:
          OP_SERVICE_ACCOUNT_TOKEN: ${{ secrets.OP_SERVICE_ACCOUNT_TOKEN }}
          SONATYPE_USERNAME: op://opensearch-infra-secrets/maven-central-portal-credentials/username
          SONATYPE_PASSWORD: op://opensearch-infra-secrets/maven-central-portal-credentials/password
```
All we need is add `onepassword` token in the secrets of this repo.

Reply:
Also the URL [here](https://github.com/opensearch-project/data-prepper/blob/main/buildSrc/src/main/groovy/data-prepper.publish.gradle#L20) needs to be updated from
`https://aws.oss.sonatype.org/content/repositories/snapshots` to `https://central.sonatype.com/repository/maven-snapshots/`

Reply:
@zelinh , @gaiksaya , I updated per your comments and following some examples from this issue:

https://github.com/opensearch-project/opensearch-build/issues/5551